src/Microsoft.Azure.ServiceBus/Message.cs

```diff
@@ -44,10 +52,10 @@ public Message(byte[] array)
         /// <remarks>
         /// The easiest way to create a new message from a string is the following:
         /// <code>
-        /// message.Body = System.Text.Encoding.UTF8.GetBytes("Message1");
+        /// message.Body = new ArrySegment&lt;byte&gt;(System.Text.Encoding.UTF8.GetBytes("Message1"));
         /// </code>
         /// </remarks>
-        public byte[] Body { get; set; }
+        public ArraySegment<byte> Body { get; set; }
```

Comment:
K. Thanks for the feedback. Yeah, I will have another discussion with team tomorrow and will update the pull request. But generally this change is mainly to change the message to use Amqp Data as we do some extra processing in both old client and service for AmqpValue and Data will be processed as is which is desired.